include/clad/Differentiator/Compatibility.h

```diff
@@ -21,6 +21,97 @@ namespace clad_compat {
 using namespace clang;
 using namespace llvm;
 
+
+//clang-18 CXXThisExpr got extra argument
+
+#if CLANG_VERSION_MAJOR > 17
+   #define CLAD_COMPAT_CLANG17_CXXThisExpr_ExtraParam DEFINE_CREATE_EXPR(CXXThisExpr, (Ctx,
+#else
+   #define CLAD_COMPAT_CLANG17_CXXThisExpr_ExtraParam DEFINE_CLONE_EXPR(CXXThisExpr, (
+#endif 
+
+//clang-18 ActOnLambdaExpr got extra argument
+
+#if CLANG_VERSION_MAJOR > 17
+   #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) /**/
+#else
+   #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) ,(V).getCurrentScope()
+#endif 
+
+//Clang 18 ArrayType::Normal -> ArraySizeModifier::Normal
+
+#if LLVM_VERSION_MAJOR < 18
+#define ArrayType_Normal ArrayType::Normal
+#else
+#define ArrayType_Normal ArraySizeModifier::Normal
+#endif
```

Comment:
```suggestion
//Clang 18 ArrayType::Normal -> ArraySizeModifier::Normal

#if LLVM_VERSION_MAJOR < 18
const auto ArraySizeModifier_Normal = ArrayType::Normal
#else
const auto ArraySizeModifier_Normal ArraySizeModifier::Normal
#endif
```
It's a good idea to avoid `#define` as much as possible.
I think it's easy here. In addition, in the code we will use `clad_compat::ArraySizeModifier_Normal`, which, apart from being closer to the current version as a name, shows that it is a matter of compatibility.